FetchMigration/README.md

```diff
@@ -1,43 +1,38 @@
+comparing indices between the two clusters and only creates index metadata (settings and mappings) that do not already 
+exist on the target cluster. Internally, the tool uses [Data Prepper](https://github.com/opensearch-project/data-prepper) 
+to migrate data for these created indices.
 
-## Parameters
+The Fetch Migration tool is implemented in Python.
+A Docker image can be built using the included [Dockerfile](./Dockerfile).
 
-The first required input to the tool is a path to a [Data Prepper](https://github.com/opensearch-project/data-prepper) pipeline YAML file, which is parsed to obtain the source and target cluster endpoints.
-The second required input is an output path to which a modified version of the pipeline YAML file is written.
-This version of the pipeline adds an index inclusion configuration to the sink, specifying only those indices that were created by the index configuration tool.
-The tool also supports several optional flags:
+## Components
 
-| Flag | Purpose |
-| ------------- | ------------- |
-| `-h, --help`    | Prints help text and exits |
-| `--report, -r`  | Prints a report of indices indicating which ones will be created, along with indices that are identical or have conflicting settings/mappings.  |
-| `--dryrun`      | Skips the actual creation of indices on the target cluster |
+The tool consists of 3 components:
+* A "metadata migration" module that handles metadata comparison between the source and target clusters. 
+This can output a human-readable report as well as a Data Prepper pipeline `yaml` file.
+* A "migration monitor" module that monitors the progress of the migration and shuts down the Data Prepper pipeline 
+once the target document count has been reached
+* An "orchestrator" module that sequences these steps as a workflow and manages the kick-off of the Data Prepper 
+process between them.
 
-### Reporting
-
-If `--report` is specified, the tool prints all processed indices organized into 3 buckets:
-* Successfully created on the target cluster
-* Skipped due to a conflict in settings/mappings
-* Skipped since the index configuration is identical on source and target
+The orchestrator module is the Docker entrypoint for the tool, though each component can be executed separately 
+via Python. Help text for each module can be printed by supplying the `-h / --help` flag.
 
 ## Current Limitations
 
-* Only supports ElasticSearch and OpenSearch endpoints for source and target
-* Only supports basic auth
-* Type mappings for legacy indices are not handled
-* Index templates and index aliases are not copied
-* Index health is not validated after creation
-
-## Usage
+* Fetch Migration runs as a single instance and does not support vertical scaling or data slicing
+* The tool does not support customizing the list of indices included for migration
+* Metadata migration only supports basic auth
+* The migration does not filter out `red` indices
```

Comment:
Why is this a limitation? what if at the time of migration, that index was red, but the user still wanted that index to be migrated?

Reply:
a `red` index means one or more primary shards are unassigned. For indices that were previously non-red, this means that some portion of the index is not readable. Thus, even though migrating such an index would go through, the data on the target index would be incomplete.